core/arch/arm/plat-k3/drivers/sa2ul_rng.c

```diff
@@ -106,6 +129,9 @@ TEE_Result hw_get_random_bytes(void *buf, size_t len)
 	uint8_t *buffer = buf;
 	size_t buffer_pos = 0;
 
+	if (!sa2ul_rng_is_enabled())
+		sa2ul_rng_init_seq();
```

Comment:
hw_get_random_bytes() may be called while there is already enough random bits in our pool, we only need to check if the TRNG is enabled when we are about to actually do a read. This check should go in sa2ul_rng_read128().

Reply:
Oh yes.
No need to check if the TRNG is enabled if the pool is not empty.
If the check is moved in sa2ul_rng_read128, it avoids lots of useless accesses.